src/pool/srv_util.c

```diff
@@ -451,3 +451,29 @@ ds_pool_check_failed_replicas(struct pool_map *map, d_rank_list_t *replicas,
 	}
 	return 0;
 }
+
+/** The caller are responsible for freeing the ranks */
```

Comment:
better to be "freeing the ranks by map_ranks_fini()", as the d_rank_list_free() cannot use